src/test/java/org/opensearch/neuralsearch/processor/TextEmbeddingProcessorTests.java

```diff
@@ -730,7 +730,7 @@ public void testBuildVectorOutput_withNestedList_successful() {
         IngestDocument ingestDocument = createNestedListIngestDocument();
         TextEmbeddingProcessor textEmbeddingProcessor = createInstanceWithNestedMapConfiguration(config);
         Map<String, Object> knnMap = textEmbeddingProcessor.buildMapWithTargetKeys(ingestDocument);
-        List<List<Float>> modelTensorList = createMockVectorResult();
+        List<List<Float>> modelTensorList = createRandomOneDimensionalMockVector(2, 2, 0.0f, 1.0f);
```

Comment:
Can you tell me why making this change?

Reply:
In former createMockVectorResult, it always return 7 results. In this test case, we only send one inference text and want single result to return. If it return more than one result, we can't trigger the IndexOutOfBoundsException, so that we should control the number of vectors it return.